### DISCUSSION TITLE
Do you have an alt node SDK with TypeScript support?

### DISCUSSION THREAD

**Discussion by haf** [swellstores/community#80] [Ideas]
What the title says?

**Comment by swelllydia** [comment]
`swell-js` version `3.19.2` was just published to npm with initial type support
https://www.npmjs.com/package/swell-js

**Reply by markus-gx** [reply]
@swelllydia You are absolutely right. Gorgeous 🥳 ; But I am not able to import the types for use in my project after removing @types/nuxt-swell and only using swell-js `3.19.2`. (e.g [Nuxt Module](https://github.com/markus-gx/nuxt-swell))

**Comment by gusfune** [comment]
The types are missing from 3.19.2 as reported in here https://github.com/swellstores/swell-js/issues/80
In the mean time you can use `@types/swell-js` until it's fixed.

**Comment by swellmike** [comment]
swell-js v3.19.3 has been published, which should include the types files within the dist.

**Comment by haf** [comment]
I'm using `swell-node`, not `swell-js` though. Any typescript for this one?

**Reply by logeshswell** [reply]
@haf I think it would take some more time. I will keep you posted when there's an update on this.

**Reply by gusfune** [reply]
You can embed your own typings locally. I started working on a `swell-node` typing, but because the way it's structured became too much of a task to maintaining alongside `swell-js` types, so I only did what I actually needed.

You can get the incomplete code, chuck into your repo and customise however you want: https://gist.github.com/gusfune/24177174e1a9ea17b7ba0b6e7ff06ffa

**Comment by haf** [comment]
@logeshswell I can give you typings if you pay for my time to do it.

**Comment by sjoerdvanbekhoven** [comment]
Any updates on this? I don't see how this can be used properly without TypeScript support

**Reply by ericingram** [reply]
Not yet, we're discussing it but likely won't come for a bit longer as we have higher priorities. Types need to be dynamically generated for each store due to the custom models.
